plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/grok/GrokProcessor.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.ingest.processor.grok;
+
+import org.elasticsearch.ingest.Data;
+import org.elasticsearch.ingest.processor.ConfigException;
+import org.elasticsearch.ingest.processor.Processor;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Properties;
+
+public final class GrokProcessor implements Processor {
+
+    public static final String TYPE = "grok";
+    public static final String PATTERNS_PATH = "/org/elasticsearch/ingest/processor/grok";
+    public static final String PATTERN_PROPERTIES = "pattern.properties";
+
+    private final String matchField;
+    private final String matchPattern;
+    private Grok grok;
+
+    public GrokProcessor(String matchField, String matchPattern) throws ConfigException {
```

Comment:
IllegalArgumentException is unchecked no need to put on the interface you can just throw it